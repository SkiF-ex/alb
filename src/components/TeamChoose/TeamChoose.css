.coose_block {
    display: flex;
    flex-direction: column;
    align-items: center;
    height: 100%;
    justify-content: center;
    font-size: 50px;
    font-style: italic;
    color: rgb(20, 118, 184);
}

.coose_block p {
    margin: 0;
}

.choose_button {
    background-color: transparent;
    border: 1px solid black;
    font-size: 30px;
    margin-top: 50px;
}