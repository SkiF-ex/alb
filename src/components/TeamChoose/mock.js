export const DATA_MOCK_WEB_TEAM = {
  title: 'WEB TEAM',
  subtitle: 'BLAZING FAST WEB',
  description: `Lorem Ipsum is simply dummy text of the printing and typesetting industry.`,
  subdescription: `Lorem Ipsum is simply dummy text of the printing and typesetting industry.`
}

export const DATA_MOCK_MOBILE_TEAM = {
  title: 'WEB MOBILE TEAM',
  subtitle: 'BLAZING FAST WEB',
  description: `Lorem Ipsum is simply dummy text of the printing and typesetting industry.`,
  subdescription: `Lorem Ipsum is simply dummy text of the printing and typesetting industry.`
}
