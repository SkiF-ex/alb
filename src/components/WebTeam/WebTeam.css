#webTeam {
    position: relative;
    height: 800px;
    display: flex;
    width: 100%;
}

.page {
    height: 800px;
    width: 100%;
    border: 1px solid black;
    margin: 0 3px;
}

.developers {
    display: flex;
    justify-content: space-around;
    margin: 10px 10px;
    flex-wrap: wrap;
}

.team_leaders {
    margin-top: 0;
}

.team_logo {
    height: 230px;
    width: 150px;
    border: 1px solid green;
    display: flex;
    justify-content: center;
    align-items: center;
}

.team_logo p {
    margin: 0;
    font-size: 20px;
    color: green;
}

.album_header {
    display: flex;
    padding: 40px 0 0 40px;
}

.team_info {
    margin-left: 20px;
}

.head_title {
    color: rgb(99, 96, 96);
    font-weight: bold;
    font-size: 25px;
    margin-top: 5px;
}

.head_item {
    color: rgb(74, 86, 252);
    font-size: 20px;
}

.head_info {
    width: 400px;
    color: red;
}
