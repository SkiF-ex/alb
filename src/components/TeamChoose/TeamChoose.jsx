import Homepage from "../Homepage";
import WebTeam from "../WebTeam/WebTeam";
import './TeamChoose.css';

const DATA_MOCK_WEB_TEAM = {
    title: 'WEB TEAM',
    subtitle: 'BLAZING FAST WEB',
    description: `Lorem Ipsum is simply dummy text of the printing and typesetting industry.`,
    subdescription: `Lorem Ipsum is simply dummy text of the printing and typesetting industry.`
}

const DATA_MOCK_MOBILE_TEAM = {
    title: 'WEB MOBILE TEAM',
    subtitle: 'BLAZING FAST WEB',
    description: `Lorem Ipsum is simply dummy text of the printing and typesetting industry.`,
    subdescription: `Lorem Ipsum is simply dummy text of the printing and typesetting industry.`
}

const TeamChoose = ({handlePage}) => {
    return (
        <>
            <button className="button_item button_previous" onClick={() => handlePage(<Homepage handlePage={handlePage}/>)}></button>
            <section id="webTeam">
                <div className="page">
                    <div className="coose_block">
                        <p>WEB TEAM</p>
                        <p>ALBUM</p>
                        <button className="choose_button" onClick={() => handlePage(<WebTeam handlePage={handlePage} team={'webprofiles'} dataMock={DATA_MOCK_WEB_TEAM}/>)}>Open</button>
                    </div>
                </div>
                <div className="page">
                    <div className="coose_block">
                        <p>MOBILE TEAM</p>
                        <p>ALBUM</p>
                        <button className="choose_button" onClick={() => handlePage(<WebTeam handlePage={handlePage} team={'mobileprofiles'} dataMock={DATA_MOCK_MOBILE_TEAM}/>)}>Open</button>
                    </div>
                </div>
            </section>
            <button className="button_hidden button_item"></button>
        </>
    );
}

export default TeamChoose;