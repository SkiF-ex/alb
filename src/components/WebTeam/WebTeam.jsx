import Profile from "../molecules/Profile";
import AlbumFirstPage from "../AlbumPages/AlbumFirstPage";
import { useEffect, useState } from "react";
import './WebTeam.css'
import AlbumSecondaryPage from "../AlbumPages/AlbumSecondaryPage";
import TeamChoose from "../TeamChoose/TeamChoose";

const WebTeam = ({handlePage, team, dataMock}) => {
    const [developers, setDevelopers] = useState('');
    const [page, setPage] = useState(1);
    const [endpoint, setEndpoint] = useState(14);

    useEffect(() => {
        fetch(`http://localhost:3004/developers`).then((response) => response.json()).then((data) => setDevelopers(data[0].team.filter(elem => elem.type === team)));
    }, []);

    const handlePager = (state) => {
        return state ? (setPage(page + 1), setEndpoint(endpoint + 18)) : (setPage(page - 1), setEndpoint(endpoint - 18));
    }

    const isPage = (page) => {
        return page < 2 ? <>
                <AlbumFirstPage info={dataMock} Profile={Profile} developers={developers}/>
                <AlbumSecondaryPage Profile={Profile} developers={developers} endpoint={endpoint}/>
            </> : <>
                <AlbumSecondaryPage Profile={Profile} developers={developers} endpoint={endpoint - 9}/>
                <AlbumSecondaryPage Profile={Profile} developers={developers} endpoint={endpoint}/>
            </>
    }
    
    const leftButtonSwitch = (page) => {
        return <button className="button_item button_previous" onClick={() => page < 2 ? handlePage(<TeamChoose handlePage={handlePage}/>) : handlePager(false)}></button>
    }

    return (
        <>
            {leftButtonSwitch(page)}
            <section id="webTeam">
                {isPage(page)}
            </section>
            {endpoint >= developers.length ? <button className="button_hidden button_item"></button> : <button className="button_item" onClick={() =>  handlePager(true)}></button>}
        </>

    );
}

export default WebTeam;