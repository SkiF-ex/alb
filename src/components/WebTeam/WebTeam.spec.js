import WebTeam from "./WebTeam";
import { useParams } from "react-router-dom";

jest.mock('react-router-dom', () => {
  return { useParams: () => ({ team: ':web' })}
});

import React from "react";
import {DATA_MOCK_MOBILE_TEAM} from "../TeamChoose/mock";
import { act, create } from 'react-test-renderer';

describe('WebTeam component', () => {
  it('should render component', () => {
    let component;
    act(() => {
      component = create(
        <WebTeam dataMock={DATA_MOCK_MOBILE_TEAM} />
      );
    });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
