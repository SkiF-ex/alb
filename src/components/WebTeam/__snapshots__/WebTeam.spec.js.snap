// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebTeam component should render component 1`] = `
Array [
  <section
    id="webTeam"
  >
    <div
      className="page"
    >
      <div
        className="album_header"
      >
        <div
          className="team_logo"
        >
          <p>
            TEAM LOGO
          </p>
        </div>
        <div
          className="team_info"
        >
          <p
            className="head_title"
          >
            WEB MOBILE TEAM
          </p>
          <p
            className="head_item"
          >
            BLAZING FAST WEB
          </p>
          <p
            className="head_info"
          >
            Lorem Ipsum is simply dummy text of the printing and typesetting industry.
          </p>
          <p
            className="head_info"
          >
            Lorem Ipsum is simply dummy text of the printing and typesetting industry.
          </p>
        </div>
      </div>
      <div
        className="developers team_leaders"
      />
      <div
        className="developers"
      />
    </div>
    <div
      className="page"
    >
      <div
        className="developers"
      />
    </div>
  </section>,
  <button
    className="button_hidden button_item button_next"
  />,
]
`;
